data-products/src/main/scala/org/sunbird/analytics/model/report/ETBMetricsModel.scala

```diff
@@ -0,0 +1,108 @@
+
+// Textbook ID, Medium, Grade, Subject, Textbook Name, Textbook Status, Created On, Last Updated On, Total content linked, Total QR codes linked to content, Total number of QR codes with no linked content, Total number of leaf nodes, Number of leaf nodes with no content
+case class etbTextbookReport(slug: String, identifier: String, name: String, medium: String, gradeLevel: String,
+                               subject:String, status: String, createdOn: String, lastUpdatedOn: String, totalContentLinked: Integer,
+                               totalQRLinked: Integer, totalQRNotLinked: Integer, leafNodesCount: Integer, leafNodeUnlinked: Integer, reportName: String)
+
+// Textbook ID, Medium, Grade, Subject, Textbook Name, Created On, Last Updated On, Total No of QR Codes, Number of QR codes with atleast 1 linked content,	Number of QR codes with no linked content, Term 1 QR Codes with no linked content, Term 2 QR Codes with no linked content
+case class dceTextbookReport(slug: String, identifier: String, name: String, medium: String, gradeLevel:String, subject: String,
+                               createdOn: String, lastUpdatedOn: String, totalQRCodes: Integer, contentLinkedQR: Integer,
+                               withoutContentQR: Integer, withoutContentT1: Integer, withoutContentT2: Integer, reportName: String)
+
+case class FinalOutput(identifier: String, etb: Option[etbTextbookReport], dce: Option[dceTextbookReport]) extends AlgoOutput with Output
+
+
+object ETBMetricsModel extends IBatchModelTemplate[Empty,Empty,FinalOutput,FinalOutput] with Serializable {
+  implicit val className: String = "org.sunbird.analytics.model.report.ETBMetricsModel"
+  override def name: String = "ETBMetricsModel"
+
+  override def preProcess(events: RDD[Empty], config: Map[String, AnyRef])(implicit sc: SparkContext, fc: FrameworkContext): RDD[Empty] = {
+    sc.emptyRDD
+  }
+
+  override def algorithm(events: RDD[Empty], config: Map[String, AnyRef])(implicit sc: SparkContext, fc: FrameworkContext): RDD[FinalOutput] = {
+    generateReports(config)
+  }
+
+  override def postProcess(events: RDD[FinalOutput], config: Map[String, AnyRef])(implicit sc: SparkContext, fc: FrameworkContext): RDD[FinalOutput] = {
+    implicit val sqlContext = new SQLContext(sc)
+    import sqlContext.implicits._
+
+    if(events.count() > 0) {
+      val configMap = config("reportConfig").asInstanceOf[Map[String, AnyRef]]
+      val reportConfig = JSONUtils.deserialize[ReportConfig](JSONUtils.serialize(configMap))
+
+      val etbTextBookReport = events.map(report => {
+        if(report.etb.size!=0) report.etb.get else etbTextbookReport("","","","","","","","","",0,0,0,0,0,"")
+      }).filter(textbook=>textbook.identifier!="")
+
+      val dceTextBookReport = events.map(report => {
+        if(report.dce.size!=0) report.dce.get else dceTextbookReport("","","","","","","","",0,0,0,0,0,"")
+      }).filter(textbook=>textbook.identifier!="")
```

Comment:
Same with this as well. Update the null/empty check

Reply:
Done